server/index.js

```diff
@@ -127,6 +127,30 @@ export default class Server {
         await this.renderJSON(req, res, pathname)
       },
 
+      '/_next/:buildId/page/:path*': async (req, res, params) => {
+        const paths = params.path || ['']
+        const pathname = `/${paths.join('/')}`
+
+        if (this.dev) {
+          await this.hotReloader.ensurePage(pathname)
+        }
+
+        if (!this.handleBuildId(params.buildId, res)) {
```

Comment:
I like this, and it makes sense, however since buildId is in the URL segment, do you think it would make more sense to 404 on a buildId mismatch? Conceptually isn't a buildId mismatch just a missing resource. I think would cause the `loadScript` `<script>` onerror to fire, and then your error emitter would pick it up.